sdk/core/azure-core/src/main/java/com/azure/android/core/http/responsepaging/PagedDataResponseRetriever.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.http.responsepaging;
+
+import com.azure.android.core.http.Response;
+import com.azure.android.core.util.paging.Page;
+
+import java.util.Objects;
+
+/**
+ * Represents a synchronized paged data provider with fixed page size that returns both page and response
+ * @param <T> type of the items in the page
+ * @param <P> type of the page
+ */
+public abstract class PagedDataResponseRetriever<T, P extends Page<T>> {
```

Comment:
Likewise if this is the sync version of `AsyncPagedDataRetriever`, let's make this `PagedDataRetriever`.

Reply:
There is a PagedDataRetriever in util.paging which does not return Response.

Reply:
Leaving as is